### ISSUE TITLE
🚀 Feature: WeChat Emoji Data

### ISSUE THREAD

**Issue by theswerd** [JoshuaKGoldberg/emoji-platform-data#145]
### Bug Report Checklist

- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/emoji-platform-data/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

In my WeChat app, I found their emoji search definitions @ `/Applications/WeChat.app/Contents/Resources/NgEmojiMap.bundle/gemoji.json`

I know they aren't a part of the original list, but seems useful.
[wechat-emojis.json](https://github.com/JoshuaKGoldberg/emoji-platform-data/files/15446835/wechat-emojis.json)

**Comment by JoshuaKGoldberg** [comment]
Yes! You'll have to forgive my locality, I don't use WeChat personally and forget it's actually used by _many_ people. Lovely to see that they've got emoji data available in JSON. Accepting PRs, thanks for filing!

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @theswerd for ideas.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙

**Comment by theswerd** [comment]
Actually found this on accident searching my Mac, I also found that the Apple emoji tags are not stored in plaintext anywhere on Macs.

**Comment by JoshuaKGoldberg** [comment]
Following up on this with the current WeChat: the `gemoji.json` this issue points at is gone, and there's no Unicode emoji metadata left in WeChat to replace it.

I installed WeChat 4.1.15 for Mac (current build) and went looking:

- **`NgEmojiMap.bundle` no longer exists.** The 4.x rewrite dropped it, and a grep of the whole 1.4 GB bundle for `thumbsup`, `octopus`, `1f419`, 🐙, 😄 and 微笑 turns up nothing.
- WeChat 4.x is Qt-based, so the data could have been compressed inside the 354 MB `wechat.dylib`. I decompressed every zlib/zstd candidate in it (29,314 of them), plus the mini program's `wmpf_resources.pak`, `Assets.car` and the icon dylib. **Not one blob contains a Unicode emoji glyph.**
- **What it ships instead is ICU.** `wechat.dylib` embeds an ICU 74.1 data package (2,235 entries) including `uemoji.icu`; extracted, that file is byte-for-byte identical to the official [`icu4c-74_1-data-bin-l`](https://github.com/unicode-org/icu/releases/tag/release-74-1) release (sha256 `26a3ef6a9204de7198522df92fb178b0268a3d300a18340a42a422a999e1a6d8`). That's why the binary references `RGI_Emoji`, `RGI_Emoji_ZWJ_Sequence` and `Emoji_Keycap_Sequence`: what counts as an emoji in WeChat is exactly what Unicode 15.1 / CLDR 44 says, and drawing is delegated to Apple Color Emoji, the system font. There's no CoreEmoji or EmojiFoundation linkage either.
- WeChat's own data is its **stickers**: a 109 entry `<emoji-panel>` (the `/::)` shortcuts and `/ue436` private use codes) and a 45 entry `<newemoji>` table with cn/qq/tw/en/th names and PNG file names — `[Smirk]`, `[Facepalm]`, `[Doge]`. The same table appears in the mini program runtime, and every UI class in the binary is `mmui::Emoticon*`. Those are images rather than Unicode emoji, so they can't be keyed by glyph the way everything in this repository is.
- The old file isn't recoverable from Tencent: `dldir1.qq.com/weixin/mac/WeChatMac.dmg` now 404s and the Sparkle appcast serves only the current build. The Windows installer is still on the 3.9.12.57 line, but its `WeChatResource.dll` and `WeChatWin.dll` have no emoji names in ASCII or UTF-16.

So the file attached here was a 2016 snapshot of [GitHub's emoji list](https://github.com/wooorm/gemoji) that WeChat 3.x for Mac happened to carry — covering 845 of the 1,913 emoji in `emoji-platform-data`, adding none of its own, and adding no search terms another source doesn't already know. Modern WeChat inherits its emoji identity from Unicode directly, so there's nothing WeChat-specific left to export.

Thanks @theswerd — this was a real find when you filed it. Unless someone knows of another WeChat surface that carries names or keywords, I think this is now not actionable. 🗝️
